pages/agent/v3/hooks.md.erb

```diff
@@ -101,12 +101,16 @@ a plugin can define.
 
 ### Creating agent lifecycle hooks
 
-Agent lifecycle hooks can be written in the programming language of your choice
-and are executed by the Buildkite agent. See the documentation for each agent
-lifecycle hook for details on the interface between them and the Buildkite
-agent.
-
+The Buildkite agent executes agent lifecycle hooks.
 These hooks can only be defined in the [agent `hooks-path`](#hook-locations-agent-hooks) directory.
+Agent lifecycle hooks can be written in any programming language, but make sure the file is executable.
```

Comment:
Slightly mixed verbs here? And we should probably say as long as the file is executable by the user the agent is running as or something for Linux like stuff?

Reply:
And would it be useful to say  "unlike job lifecycle hooks, swhich are sourced" here?

Reply:
Verb mismatch fixed with 6b9ee458a1e9cfa1db77956cfdd282d8e731d431. Unix-like permissions noted with 01505466a7d8125ba15464df8535ea1c4c34573b.

I'm of two minds about mentioning job lifecycle hooks' source behavior here. I think executing a script as a script is fairly typical, while sourcing isn't. If you're reading this _before_ being exposed to the job hooks, it's a bit mysterious and weird. As a reference, it's worth mentioning (e.g., if you're landing on this section after already doing some work with job hooks), though I'm struggling with how to say so without implying that the other one is, in fact, weird.

For now I've left it without mention. Thoughts?

Reply:
I'll buy that.